lib/astutils.cpp

```diff
@@ -628,7 +628,8 @@ static const Token * followVariableExpression(const Token * tok, bool cpp, const
     const Token * lastTok = precedes(tok, end) ? end : tok;
     // If this is in a loop then check if variables are modified in the entire scope
     const Token * endToken = (isInLoopCondition(tok) || isInLoopCondition(varTok) || var->scope() != tok->scope()) ? var->scope()->bodyEnd : lastTok;
-    if (!var->isConst() && (!precedes(varTok, endToken) || isVariableChanged(varTok, endToken, tok->varId(), false, nullptr, cpp)))
+    if (!var->isConst() && (!precedes(varTok, endToken) || isVariableChanged(varTok, endToken, tok->varId(), false, nullptr, cpp) ||
+                            isVariableChanged(varTok, endToken, varTok->varId(), false, nullptr, cpp)))
```

Comment:
varTok could be an expression, so this wont always work correctly. I already updated in PR #2805 to check if the expression is modified. This may fix your issue already.

Reply:
Indeed, it seems that #2805 fixes this. I'll close this pull-request.
Many thanks !